src/proto/tagpost.proto

```diff
@@ -7,17 +7,68 @@ option java_package = "com.google.tagpost";
 option java_outer_classname = "TagPostProto";
 
 service TagpostService {
+  // dummy rpc service
   rpc FetchMessage(FetchMessageRequest) returns (FetchMessageResponse) {}
+  // fetch all threads by primary tag
+  rpc FetchThreadsByTag(FetchThreadsByTagRequest) returns (FetchThreadsByTagResponse) {}
+  // fetch all comments under one thread
+  rpc FetchCommentsUnderThread(FetchCommentsUnderThreadRequest) returns (FetchCommentsUnderThreadResponse) {}
+  // fetch all comments by a set of tags
+  rpc FetchCommentsByTags(FetchCommentsByTagsRequest) returns (FetchCommentsByTagsResponse) {}
```

Comment:
Does this also include non-primary tags? If so we may need some modification with database schema.